backend_new/test/unit/services/application.service.spec.ts

```diff
@@ -407,4 +410,1472 @@ describe('Testing application service', () => {
+        emailAddress: 'example@email.com',
+        address: exampleAddress,
+      },
+      applicationsAlternateAddress: exampleAddress,
+      applicationsMailingAddress: exampleAddress,
+      listings: {
+        id: randomUUID(),
+      },
+      demographics: {
+        ethnicity: 'example ethnicity',
+        gender: 'example gender',
+        sexualOrientation: 'example sexual orientation',
+        howDidYouHear: ['example how did you hear'],
+        race: ['example race'],
+      },
+      preferredUnitTypes: [
+        {
+          id: randomUUID(),
+        },
+      ],
+      householdMember: [
+        {
+          orderId: 0,
+          firstName: 'example first name',
+          middleName: 'example middle name',
+          lastName: 'example last name',
+          birthMonth: '12',
+          birthDay: '17',
+          birthYear: '1993',
+          sameAddress: YesNoEnum.yes,
+          relationship: 'example relationship',
+          workInRegion: YesNoEnum.yes,
+          householdMemberWorkAddress: exampleAddress,
+          householdMemberAddress: exampleAddress,
+        },
+      ],
+      appUrl: 'http://www.example.com',
+      additionalPhone: true,
+      additionalPhoneNumber: '111-111-1111',
+      additionalPhoneNumberType: 'example additional phone number type',
+      householdSize: 2,
+      housingStatus: 'example housing status',
+      sendMailToMailingAddress: true,
+      householdExpectingChanges: false,
+      householdStudent: false,
+      incomeVouchers: false,
+      income: '36000',
+      incomePeriod: IncomePeriodEnum.perYear,
+      language: LanguagesEnum.en,
+      acceptedTerms: true,
+      submissionDate: new Date(),
+      reviewStatus: ApplicationReviewStatusEnum.valid,
+      programs: [
+        {
+          key: 'example key',
+          claimed: true,
+          options: [
+            {
+              key: 'example key',
+              checked: true,
+              extraData: [
+                {
+                  type: InputType.boolean,
+                  key: 'example key',
+                  value: true,
+                },
+              ],
+            },
+          ],
+        },
+      ],
+    };
+
+    await expect(
+      async () => await service.create(dto, true),
+    ).rejects.toThrowError('Listing is not open for application submission');
+
+    expect(prisma.listings.findUnique).toHaveBeenCalledWith({
+      where: {
+        id: expect.anything(),
+      },
+    });
+
+    expect(prisma.applications.create).not.toHaveBeenCalled();
+  });
+
+  it('should create an application from partner site', async () => {
+    prisma.listings.findUnique = jest.fn().mockResolvedValue({
+      id: randomUUID(),
+    });
+
+    prisma.applications.create = jest.fn().mockResolvedValue({
+      id: randomUUID(),
+    });
+
+    const exampleAddress = addressFactory() as AddressUpdate;
+    const dto: ApplicationCreate = {
+      contactPreferences: ['example contact preference'],
```

Comment:
There is a lot of duplication when creating these ApplicationCreate DTOs. Could it be pulled into one place?

Reply:
cleaned it up a lil, lemme know what you think!